src/librustc_mir/transform/qualify_consts.rs

```diff
@@ -922,7 +1327,9 @@ This does not pose a problem by itself because they can't be accessed directly."
                         // this doesn't come from a macro that has #[allow_internal_unstable]
                         !self.span.allows_unstable()
                     {
-                        self.qualif = Qualif::NOT_CONST;
+                        self.mut_interior_checker.mut_interior = false;
+                        self.drop_checker.needs_drop = false;
```

Comment:
You probably don't need to touch the other flags here.

Reply:
Are you sure? I did a literal translation of course, because I wasn't sure...